### PR TITLE
BaSP-MR-117: Move get and delete logic for Classes

### PR DESCRIPTION
**Pull request by AleMac17**
Replace the data from the local state

Move the logic of the get to the thunk file in the classes folder and apply it to the entity index file.

Move the logic of the delete to the thunk file in the classes folder and apply it to the entity index file.

To make the user experience more pleasant, for each request, a spinner (isLoading) should be implemented so that the user understands that a request is being made, for example when loading the table.

### CODE REVIEW

**Comment by AleMac17** [comment]
[[Classes] - Move get and delete logic](https://trello.com/c/vjwEh44F/117-classes-move-get-and-delete-logic)

File: src/Components/Classes/index.jsx [hunk 1]
```diff
@@ -56,13 +35,17 @@ const Classes = () => {
 
   const handleConfirmDeleteClass = async () => {
     setShowDeleteWarning(false);
+    let data;
     try {
-      await deleteClasses(classToDelete);
-      setClasses((prevClasses) =>
-        prevClasses.filter((classItem) => classItem._id !== classToDelete)
-      );
+      data = dispatch(deleteClass(classToDelete));
```
Comment (EmaBaldor) [inline src/Components/Classes/index.jsx]: There should be an 'await' here.

**Review by fernandofernandezuy — APPROVED** [review]
Looks good to me!

**Review by EmaBaldor — APPROVED** [review]
Good Job!

**Review by Federico-Peralta — APPROVED** [review]
Tested, all good!

**Review by marcohzo — APPROVED** [review]

**Comment by nazarenapantanetti** [comment]
Tested in master, change requested! Follow the link to see the report: https://trello.com/c/S2tufRsy/105-classes-bug-spinner-fails

**Comment by nazarenapantanetti** [comment]
Tested in master, changes requested! Follow the link to see the complete report: https://trello.com/c/O4JGoPBW/141-classes-bug-defective-edition

**Pull request merged**